src/all/apt/String.cpp

```diff
@@ -317,27 +317,33 @@ StringBase::StringBase(StringBase&& _rhs_)
 	, m_capacity(_rhs_.m_capacity)
 	, m_length(_rhs_.m_length)
 {
-	if (_rhs_.isLocal()) {
+	if (_rhs_.isLocal())
 		strncpy(m_buf, _rhs_.m_buf, _rhs_.m_length + 1);
-	} else {
+	else
 		m_buf = _rhs_.m_buf;
-		_rhs_.m_buf = nullptr;
-		_rhs_.m_capacity = 0;
-		_rhs_.m_length = 0;
-	}
+	m_capacity = _rhs_.m_capacity;
+	m_length = _rhs_.m_length;
+	_rhs_.m_buf = nullptr;
+	_rhs_.m_capacity = 0;
+	_rhs_.m_length = 0;
 }
 StringBase& StringBase::operator=(StringBase&& _rhs_)
 {
 	if (&_rhs_ != this) {
-		if (_rhs_.isLocal()) {
```

Comment:
`if (_rhs_.isLocal())` 
If the incoming string (`_rhs_`) is local, we know that the destination string (`this`) *must* have enough storage - this is enforced by the templated classes (see String.h line 123).

In other words: `this` and `_rhs_` are the same type, and we only ever grow the storage, so we know that if `_rhs_` is local we have enough storage in `this` to do the `strncpy` below. Avoiding memory allocations/deallocations is better than saving a few bytes in this case.